src/microbe_stage/Patch.cs

```diff
@@ -16,6 +16,9 @@
 [UseThriveSerializer]
 public class Patch
 {
+    [JsonProperty]
+    public List<WorldEffectVisuals> ActiveDisplayVisuals;
```

Comment:
I think this should be renamed *and* made private and read-only. Then methods like add event and a method for this patch to send the active events for rendering / text display should be added. That way the internal details are nicely encapsulated and can be extended in the future and we avoid extra memory allocations (which would be necessary if a method like `GetActiveEvents` was created).

Reply:
How read-only if we want to change it via methods?

Reply:
I mean `readonly` so that the list object cannot be reassigned, but its contents will still be modifiable.